src/main/java/com/faforever/client/util/TimeService.java

```diff
@@ -47,13 +47,24 @@ public String asDate(TemporalAccessor temporalAccessor, FormatStyle formatStyle)
     if (temporalAccessor == null) {
       return i18n.get("noDateAvailable");
     }
-    return DateTimeFormatter.ofLocalizedDate(formatStyle)
-        .withLocale(getCurrentDateLocale())
+
+    DateInfo dateInfo = localizationPrefs.getDateFormat();
+    DateTimeFormatter formatter;
+
+    if (dateInfo.equals(DateInfo.DAY_MONTH_YEAR)) {
+      formatter = DateTimeFormatter.ofPattern(getDatePatternForDayMonthYear(formatStyle));
+    } else if (dateInfo.equals(DateInfo.MONTH_DAY_YEAR)) {
+      formatter = DateTimeFormatter.ofPattern(getDatePatternForMonthDayYear(formatStyle));
+    } else {
+      formatter = DateTimeFormatter.ofLocalizedDate(formatStyle);
+    }
```

Comment:
This can be a switch expression

Reply:
Is this requirement or suggestion?

For me it has same readability, moreover less code as no useless break statement and no default state(which shouldn't even exists in 1st place). But hey, if you want it that way. I don't mind.

Reply:
This is how it would look like
```
    DateTimeFormatter formatter = switch(dateInfo) {
      case DateInfo.DAY_MONTH_YEAR -> DateTimeFormatter.ofPattern(getDatePatternForDayMonthYear(formatStyle))
      case DateInfo.MONTH_DAY_YEAR -> DateTimeFormatter.ofPattern(getDatePatternForMonthDayYear(formatStyle));
      default ->
        DateTimeFormatter.ofLocalizedDate(formatStyle);
    }
 ```

- no dangling variable declaration
- only a single assignment
- no .equals calls

This is more readable.

Reply:
@Brutus5000 is default necessary? I mean from code-style perspective @Sheikah45 mentioned in other issue to use exhaustive switch .. I'm trying to find out, how the code-style should look like here ..

Reply:
Yes I would add the auto case instead of just handling it in default.

Reply:
So it just looks like this

```
    DateTimeFormatter formatter = switch(dateInfo) {
      case DAY_MONTH_YEAR -> DateTimeFormatter.ofPattern(getDatePatternForDayMonthYear(formatStyle))
      case MONTH_DAY_YEAR -> DateTimeFormatter.ofPattern(getDatePatternForMonthDayYear(formatStyle));
      case AUTO -> DateTimeFormatter.ofLocalizedDate(formatStyle);
    }
 ```

Reply:
Also, we could just make the function return a datetimeformatter instead of just the pattern string.

Then we could also make those constants so we don't recreate them all the time.

Reply:
@Sheikah45  fine for me.

> Also, we could just make the function return a datetimeformatter instead of just the pattern string.

I'm not quite sure if I understand you(could you elaborate more on it?), but I doubt there is other way than to use pattern. Already in just Date locale, every locale has its own way how is **short, medium, long, full** style written US is different to FR, etc.. And currently, what I have written (using date any custom format) will differ to auto format for at least FR.

Example of France FormatStyle.Medium auto:
> 9 mai 2022

Example of France FormatStyle.Medium DayMonthYear:
> mai 9, 2022

Example of France FormatStyle.Medium DayMonthYear:
> 9 mai, 2022

As u can see France auto format doesn't use comma, however US style use it. Across all other locale it is more inconsistent. Moreover, if you add more complex formats (long, full - this is reason I avoided full style, while it is not used anywhere), maybe even dateTime itself, the more inconsistencies arise.

A solution would be to agree, which format we should follow for every locale. It seems to me too much of trouble for minimal return.

Reply:
In the functions getDatePatternForX you are returning a string that you just feed into DateTimeFormatter.ofPattern

Instead we can just return the DateTimeFormatter from the getDatePatternForX and rename it to getFormatterForX.

And then instead of creating the datetimeformatter each time we can just define private static final members in the class and return those.

Reply:
I am only talking about returning a date time format err from the getPatternForX methods. Those return constant strings so we can just make them return constant datetime formatters.

There would be no restart necessary and it would still work just changing the formatstyle during runtime

Reply:
I hear and I obey.